### PR TITLE
recall/ops: support graceful runner shutdown

### PR DESCRIPTION
**Pull request by idy**
## Summary
- preserve worker failures when GracefulStop interrupts error backoff
- release supervisor child contexts on successful GracefulStop
- avoid unbounded waiting after GracefulStop deadline cancellation
- add recall.NamespaceFor contract tests for global, user, sanitized, empty, and delimiter-containing scopes

## Tests
- go test ./memory/recall/ops ./memory/recall
- go test -race ./memory/recall/ops

### CODE REVIEW

File: memory/recall/ops/runner.go [hunk 1]
```diff
@@ -69,15 +76,18 @@ func (r *Runner) Run(ctx context.Context, opts RunOptions) error {
 			if errdefs.IsValidation(err) {
 				return err
 			}
-			if waitErr := sleepContext(ctx, r.cfg.ErrorBackoff); waitErr != nil {
+			if waitErr := sleepContext(ctx, stop, r.cfg.ErrorBackoff); waitErr != nil {
```
Comment (lIang70) [inline memory/recall/ops/runner.go:79]: This can turn a failed in-flight drain into a successful graceful stop. If `GracefulStop` closes `stop` while `Drain` returns a non-validation error, this `sleepContext` wakes on `stop` and returns `nil`; the loop then continues, hits `stopped(stop)`, and returns `nil`, so `Start` records no worker error. Please preserve and return the original drain error when stop is observed during error backoff.

File: memory/recall/ops/supervisor.go [hunk 2]
```diff
@@ -47,8 +49,42 @@ func (s *Supervisor) Stop() error {
...
+		close(s.stop)
+	})
+	done := make(chan struct{})
+	go func() {
+		s.wg.Wait()
+		close(done)
+	}()
+	select {
+	case <-done:
+		return s.err()
```
Comment (lIang70) [inline memory/recall/ops/supervisor.go]: The successful graceful-stop path never calls `s.cancel()`, even though `Start` created a child context with `context.WithCancel`. `Stop` releases that context, but repeated successful `GracefulStop` lifecycles under a long-lived parent can retain context state until the parent is cancelled. Please call `s.cancel()` before returning on the success path.

File: memory/recall/ops/supervisor.go [hunk 3]
```diff
@@ -47,8 +49,42 @@ func (s *Supervisor) Stop() error {
...
+	case <-ctx.Done():
+		s.cancel()
+		s.wg.Wait()
```
Comment (lIang70) [inline memory/recall/ops/supervisor.go]: `ctx` looks like the caller's shutdown deadline, but once it fires this still waits on `s.wg.Wait()` without any bound. If a processor is stuck in an external call or otherwise ignores context cancellation, `GracefulStop(ctx)` can hang forever despite the deadline. Please either return `ctx.Err()` after issuing `s.cancel()`, or add an explicit bounded forced-stop phase.

File: memory/recall/namespace.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+import (
+	"strconv"
+	"strings"
+
+	retrievalns "github.com/GizClaw/flowcraft/memory/retrieval/namespace"
+)
+
+// NamespaceFor returns the v2 retrieval namespace for a memory scope.
+func NamespaceFor(s Scope) string {
```
Comment (lIang70) [inline memory/recall/namespace.go:11]: Since this is a new public helper used to derive retrieval namespace strings, please add direct table-driven tests for the wrapper contract: global scope, user scope, sanitized runtime/user IDs, empty runtime/user behavior, and delimiter-containing users like `bob__u_alice`. The existing `retrieval/namespace` tests cover the generic encoder, but they do not pin this public `recall.NamespaceFor` API to the v2 `recall` namespace.

**Review by lIang70 — COMMENTED** [review]
Reviewed the shutdown changes and the new namespace helper. The targeted tests pass (`go test ./memory/recall/ops ./memory/recall` and `go test -race ./memory/recall/ops`), but I found a few shutdown semantics issues that should be addressed before merge.
